ncl/ncl_entries/general_applied_math.ipynb

```diff
@@ -0,0 +1,713 @@
+{
```

Comment:
<p>I'm not sure this is the case for rounding / truncating. It looks like the behavior is a little different between NCL and Python (standard lib and numpy), but there's a little more nuance to it:</p><p><a href="https://docs.python.org/3/library/functions.html#round" target="_blank">https://docs.python.org/3/library/functions.html#round</a></p><p><a href="https://numpy.org/doc/stable/reference/generated/numpy.round.html" target="_blank">https://numpy.org/doc/stable/reference/generated/numpy.round.html</a></p><p><a href="https://www.ncl.ucar.edu/Document/Functions/Built-in/round.shtml" target="_blank">https://www.ncl.ucar.edu/Document/Functions/Built-in/round.shtml</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NCAR/geocat-applications/pull/77/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

Reply:
<p>+1 here.</p><p> </p><p>I also think the language around <code>math.ceil</code> being "rounding up" and <code>math.floor</code> being "rounding down" could be clarified. I personally don't think of ceiling and floor functions as rounding, but I'm interested in hearing others' thoughts here.</p>

Reply:
I'm ok with the "round up" "round down" language.  It is a bit confusing here with so many definitions of round, but I think it's in the Python docs with similar language and I don't have a better suggestion offhand.  
 
I was more concerned that this statement is confusing and/or needs more clarification: 
"NCL distinguishes between rounding a decimal value and truncating it, but Python rounds the values by default, while truncating requires converting to a string" 

And that this statement isn't true in some cases "The Python function round function rounds a float up when 5 or greater and rounds down when 4 or lower."  For both math and numpy rounding is done toward the even choice so this switches back and forth.

It's probably also worth noting that the NCL round function only rounds to the nearest whole number, which is considerably less flexible than the Python implementations.

I do think the differences sections are a little tricky in general and this function in particular is surprisingly quite different in NCL from the Python ones.  

I don't know that this has to be perfect in terms of language and don't want to hold this up too much before merging, but I am worried about the correctness at least.

Reply:
I see what you mean. So, NCL function this is mean to cover is [decimalPlaces](https://www.ncl.ucar.edu/Document/Functions/Contributed/decimalPlaces.shtml), which either truncates or rounds the values.  decimalPlaces rounds/truncates a value to a specific number of decimal places

Meanwhile, NCL [round](https://www.ncl.ucar.edu/Document/Functions/Built-in/round.shtml) function rounds the float/double to nearest whole number. The round function isn't currently being covered here, but will be as part of v2 (https://github.com/NCAR/geocat-applications/issues/79). I can clear up the language here though

Reply:
This looks better overall, but it looks like the description of the Python round function still needs an update.

> And that this statement isn't true in some cases "The Python function round function rounds a float up when 5 or greater and rounds down when 4 or lower." For both math and numpy rounding is done toward the even choice so this switches back and forth.

Reply:
I reworded it a bit and added a note based on a disclaimer in the Python round function